presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcPageSource.java

```diff
@@ -57,154 +45,92 @@
     public static final int BUCKET_NUMBER_COLUMN = -4;
 
     private final Optional<ShardRewriter> shardRewriter;
-
-    private final OrcBatchRecordReader recordReader;
-    private final OrcDataSource orcDataSource;
-
     private final BitSet rowsToDelete;
 
-    private final List<Long> columnIds;
-    private final List<Type> types;
-
-    private final Block[] constantBlocks;
-    private final int[] columnIndexes;
-
-    private final AggregatedMemoryContext systemMemoryContext;
-
-    private int batchId;
-    private long completedPositions;
-    private boolean closed;
+    private OrcReadPageSource parentPageSource;
```

Comment:
This is not the right design. This file should remain (almost) unchanged. The only critical change to be made is on `recordReader`. That should be replaced by another page source (which inherits `ConnectorPageSource`) to represent the source data. Let me know if you want to chat in person.

Reply:
Can we chat in person today?